From 023ec5c0eafe8dfb5b71e400cb0c2c337fb8c108 Mon Sep 17 00:00:00 2001
From: Imran Desai <imran.desai@intel.com>
Date: Sun, 29 Mar 2020 10:49:12 -0700
Subject: [PATCH] tools/tpm2_nvreadpublic: Fix resource leak

Signed-off-by: Imran Desai <imran.desai@intel.com>
---
 tools/tpm2_nvreadpublic.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/tpm2_nvreadpublic.c b/tools/tpm2_nvreadpublic.c
index 7f8e46cbf863..eca1fd715a79 100644
--- a/tools/tpm2_nvreadpublic.c
+++ b/tools/tpm2_nvreadpublic.c
@@ -41,6 +41,7 @@ static tool_rc print_nv_public(ESYS_CONTEXT *context, TPMI_RH_NV_INDEX index, TP
     rc = tpm2_tr_get_name(context, tr_handle,
             &name);
     if (rc != tool_rc_success) {
+        free(attrs);
         return rc;
     }
 
-- 
2.31.0

